File: java/07_pgm_thread/src/co/edu/campusucc/poo/MainThreadExtends.java
package co.edu.campusucc.poo;
import java.util.Random;

import co.edu.campusucc.poo.domain.ThreadExtends;

public class MainThreadExtends{

    public static void main(String[] args) {
        ThreadExtends thread = new ThreadExtends();
        thread.setTimes(new Random().nextInt(50));
        // Invoking Thread using start() method
        thread.start();
    }
}

File: c++/07_pgm_thread/src/co/edu/campusucc/poo/MainThreadExtends.cpp
#include <iostream>
#include <thread>
#include <random>
#include "ThreadExtends.h"

using namespace std;

int main() {
ThreadExtends thread;
thread.setTimes(rand() % 50); // Generating a random number between 0 and 49
thread.start(); // Invoking thread using start() method
return 0;
}